### ISSUE TITLE
OS DirectoryCreationFailed when run as www-data?

### ISSUE THREAD

**Issue by Eir-nya** [LSchallot/JellyRoller#36]
Hi, I understand this might be far from the expected use case, but I still figured I'd post.

I'm trying to set up JellyRoller in conjunction with a custom site running php. I've set up a button to create a user using php's `exec` function. But when running *any* JellyRoller command, I get this output:

```
Error: DirectoryCreationFailed(Os { code: 13, kind: PermissionDenied, message: "Permission denied" })
```

I've already set `www-data` (user that runs the php service) as the owner of both the executable and the folder it's in. This happens when running any JellyRoller command, including `-h` or `-v`, or even without arguments. I'm not creating a directory as any part of this. Is JellyRoller trying to create a directory when run?

**Comment by LSchallot** [comment]
JellyRoller does create a directory when executed to store its configuration files.  This is a default part of confy (the library used to store the configuration).  Something I could look into is adding the ability to specify a configuration file.  This would allow you to store the configuration file alongside the JellyRoller executable.

**Comment by Eir-nya** [comment]
That would be great! In the meantime, here's a workaround:

```
XDG_CONFIG_HOME=/path/to/custom/jellyroller/directory jellyroller list-users
```

This causes the jellyroller config to be generated/saved at `/path/to/custom/jellyroller/directory/jellyroller/jellyroller.toml`.
Support for specifying a config would still be cool though :)

**Comment by LSchallot** [comment]
Ability to use a JellyRoller.config file that lives alongside the executable has been added to the project.  It will be included in the 0.5.0 release.
